### PR TITLE
Send error data back to catalog for permanent failures and update status accordingly.

### PR DESCRIPTION
**Pull request by andymutton**

### CODE REVIEW

File: analysis/src/main.js [hunk 1]
```diff
@@ -84,6 +85,8 @@ function processTasks() {
         } else {
           Ana.fail("main/processTasks");
           res.status(200).send();
```
Comment (dstockwell) [inline analysis/src/main.js]: does this still happen if line 89 fails?

File: src/manage.py [hunk 2]
```diff
@@ -650,14 +650,14 @@ def handle_post(self):
     owner = attributes['owner']
     repo = attributes['repo']
     version = attributes['version']
+    error = attributes['error'] if 'error' in attributes else None
```
Comment (dstockwell) [inline src/manage.py]: attributes.get('error', None)

File: analysis/src/main.js (hunk 1 above)
Comment (andymutton) [inline analysis/src/main.js]: Yup. That's why I ordered it this way.

Comment (dstockwell) [inline analysis/src/main.js]: But shouldn't we retry until we're able to send the result back?

**Pull request merged**